x-pack/plugins/security_solution/server/endpoint/services/actions/action_files.ts

```diff
@@ -133,3 +135,84 @@ const doesFileHaveChunks = async (
 
   return Boolean((chunks.hits?.total as SearchTotalHits)?.value);
 };
+
+/**
+ * Returns subset of fileIds that don't have any file chunks
+ *
+ * @param esClient
+ * @param fileIds
+ * @returns fileIds
+ */
+export async function fileIdsWithoutChunks(
+  esClient: ElasticsearchClient,
+  fileIds: string[]
+): Promise<string[]> {
+  const noChunkFileIds = new Set(fileIds);
+
+  const chunks = await esClient.search<{ bid: string }>({
+    index: FILE_STORAGE_DATA_INDEX,
+    body: {
+      query: {
+        terms: {
+          'bid.keyword': fileIds,
+        },
+      },
+      _source: ['bid'],
+    },
+  });
+
+  chunks.hits.hits.forEach((hit) => {
+    const fileId = hit._source?.bid;
+    if (!fileId) return;
+    noChunkFileIds.delete(fileId);
+  });
+
+  return Array.from(noChunkFileIds);
+}
+
+/**
+ * Gets files with status of READY
+ *
+ * @param esClient
+ * @param status
+ */
+export function getFilesByStatus(esClient: ElasticsearchClient, status: FileStatus = 'READY') {
+  return esClient.search({
+    index: FILE_STORAGE_METADATA_INDEX,
+    body: {
+      query: {
+        term: {
+          'file.Status.keyword': status,
+        },
+      },
+      _source: false,
+    },
+    ignore_unavailable: true,
+  });
+}
+
+/**
+ * Updates given fileIds to DELETED status
+ *
+ * @param esClient
+ * @param fileIds
+ */
+export function updateFilesStatus(
+  esClient: ElasticsearchClient,
+  fileIds: string[],
+  status: FileStatus
```

Comment:
should this be `DELETED` by default?

Reply:
If not, the comment above should be updated

Reply:
ah, good catch. will update comments, including `getFilesByStatus`.

Reply:
also,`@param status` is missing right?